zk_ee/src/system/mod.rs

```diff
@@ -242,22 +242,26 @@ where
     pub fn try_begin_next_tx(
         &mut self,
         tx_write_iter: &mut impl crate::oracle::SafeUsizeWritable,
-    ) -> Result<Option<usize>, ()> {
+    ) -> Result<Option<usize>, InternalError> {
         let next_tx_len_bytes = match self.io.oracle().try_begin_next_tx() {
             None => return Ok(None),
             Some(size) => size.get() as usize,
         };
+        // Check to avoid usize overflow in 32-bit target.
+        if next_tx_len_bytes > u32::MAX as usize - (core::mem::size_of::<u32>() - 1) {
+            return Err(internal_error!("TX length is too large"));
```

Comment:
Should we rely on the server/l1 to protect against this? Or should we handle it better? @AntonD3 @0xVolosnikov

Reply:
Pushed the fix